starknet_devnet/origin.py

```diff
@@ -63,21 +71,28 @@ class NullOrigin(Origin):
 
     def get_transaction_status(self, transaction_hash: str):
         return {
-            "tx_status": TxStatus.NOT_RECEIVED.name
+            "tx_status": TransactionStatus.NOT_RECEIVED.name
         }
 
-    def get_transaction(self, transaction_hash: str):
-        return {
-            "status": TxStatus.NOT_RECEIVED.name
-        }
-
-    def get_transaction_receipt(self, transaction_hash: str):
-        return {
-            "l2_to_l1_messages": [],
-            "status": TxStatus.NOT_RECEIVED.name,
-            "transaction_hash": transaction_hash,
-            "events": []
-        }
+    def get_transaction(self, transaction_hash: str) -> TransactionInfo:
+        return TransactionInfo.create(
+            status=TransactionStatus.NOT_RECEIVED,
+        )
+
+    def get_transaction_receipt(self, transaction_hash: str) -> TransactionReceipt:
+        return TransactionReceipt(
```

Comment:
Would it be faster to have this object created in advance?

Reply:
It would be slower for users that never call `get_transaction_receipt`